### PR TITLE
CyberArk Implementation Guide

### PR DESCRIPTION
**Pull request by kashifzeeshan**
CyberArk Implementation Gudie

### CODE REVIEW

File: advocacy_docs/partner_docs/Cyberark/03-SolutionSummary.mdx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: 'Solution Summary'
+description: 'Explanation of the solution and its purpose'
+---
+
+CyberArk is an Identity and Access Management (IAM) security tool you can use as a privileged access management tool. It offers comprehensive solutions to store, manage, and share passwords across your organizations. With highly customized security roadmaps, you can protect your company from the malware and other security threats associated with hacking. It’s a new class of custom security solutions that allows you to respond to such incidents.
+
+In addition to tracking user credentials quickly and efficiently, CyberArk prevents those credentials from being lost or corrupted. By stripping away unnecessary complexities associated with monitoring, CyberArk helps businesses avoid many of the most common data breach scenarios that start with compromised credentials.
+
+CyberArk supports EDB Postgres Advanced Server and PostgreSQL databases and store and manage passwords which help to secure data against cybersecurity attacks, malware, and hacking threats.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/03-SolutionSummary.mdx:10]: Would reword like:
CyberArk supports EDB Postgres Advanced Server and PostgreSQL databases to store and manage passwords, which helps to secure data against cybersecurity attacks, malware, and hacking threats.

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+---
+title: 'Configuration'
+description: 'Walkthrough on configuring the integration'
+---
+
+## Prerequisites
+
+Implementing CyberArk CPM (Central Policy Manager) with EDB Postgres Advanced Server requires the following components:
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:8]: There are no components listed after the colon.

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Configuring your EDB Postgres Advanced Server
+
+Following component is needed before integrating EDB Postgres Advanced Server with CyberArk CPM (Central Policy Manager):
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:12]: Would reword:
 The following component is needed before integrating EDB Postgres Advanced Server with CyberArk CPM (Central Policy Manager):

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+1. One running instance of EDB Postgres Advanced Server.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:14]: If this is also for Postgres should we say something like "A running instance of EPAS or Postgresql"?

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## Configuring CyberAr CPM (Central Policy Manager)
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:16]: CyberArk

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+Following configurations are need on the CyberArk machine before integrating CyberArk CPM (Central Policy Manager) with EDB Postgres Advanced Server:
+
+### Install PostgreSQL ODBC Driver
+
+1. Download the latest 32 bit PostgreSQL ODBC driver from the psqlodbc download site.
+2. Unzip the file.
+3. Run the msi installer and accept the defaults.
+4. Open the 32 bit ODBC Administrator from Widnows --> Run.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:25]: Windows

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+![Data Source](Images/savedatasource.png)
+
+### Setup of PostgreSQL Plugin from CyberArk Marketplace
+ 
+1. Download the PostgreSQL plugin from CyberArk Marketplace.
+2. Login to Password Vault Web Access (PVWA) interface.
+3. Open Administration --> Platform Management.
+4. Add new platform by clicking on Import Platform.
+5. Import the postgresql platform which is download from Marketplace.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:42]: downloaded

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 8]
```diff
@@ -0,0 +1,109 @@
+---
+title: 'Using'
+description: 'Walkthrough of example usage scenarios'
+---
+
+After you have configured your EDB Postgres Advanced Server and CyberArk, as stated in the Configuring section, you will be able to then use CyberArk (CPM Central Policy Manager) functionality.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:6]: would say:
you will be able to then use the CyberArk...etc.

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 9]
```diff
@@ -0,0 +1,109 @@
+---
+title: 'Using'
+description: 'Walkthrough of example usage scenarios'
+---
+
+After you have configured your EDB Postgres Advanced Server and CyberArk, as stated in the Configuring section, you will be able to then use CyberArk (CPM Central Policy Manager) functionality. 
+
+Authorized users can change passwords that are stored in the Safe through the PVWA (Password Vault Web Access). These passwords can be changed manually or replaced by a new password that is randomly generated by the Central Policy Manager. The CPM generates unique and highly secure passwords using the password policy and the random password generation mechanism.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:8]: Would add (CPM) after the first mention of Central Policy Manager so when we say CPM throughout the guide we know what it is for.

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 10]
```diff
@@ -0,0 +1,109 @@
+---
+title: 'Using'
+description: 'Walkthrough of example usage scenarios'
+---
+
+After you have configured your EDB Postgres Advanced Server and CyberArk, as stated in the Configuring section, you will be able to then use CyberArk (CPM Central Policy Manager) functionality. 
+
+Authorized users can change passwords that are stored in the Safe through the PVWA (Password Vault Web Access). These passwords can be changed manually or replaced by a new password that is randomly generated by the Central Policy Manager. The CPM generates unique and highly secure passwords using the password policy and the random password generation mechanism. 
+
+The following example demonstrate generating unique and highly secure passwords based on the password policy by CPM (CPM Central Policy Manager).
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:10]: demonstrates

Comment (jkitchens32) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:10]: by the CPM.

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+- Configure Password Policy for PostgreSQL Platform
+  - Configure password policy by setting the Password Management parameters for PostgreSQL platform
+- Create the Account for EDB Postgres Advanced Server running instance
+  - Setup Account on PVWA (Password Vault Web Access) for the EDB Postgres Advanced Server running instance, the password will be managed for this EDB Postgres Advanced Server running instance
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:15]: Set up

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+- Change the Password 
+  - Using CPM (CPM Central Policy Manager) change the password for the EDB Postgres Advanced Server running instance
+- Verify that User can login to EDB Postgres Advanced Server
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:18]: Verify that the user

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  - After the password is changed by CPM (CPM Central Policy Manager), login to EDB Postgres Advanced Server with the changed password.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:19]: Would take out the . after this sentence as we don't have it anywhere else in these bullets.

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+2. Click Administration option, and then select Platform Management. A list of supported target account platforms appears.
+
+3. Select Databases —> PostgreSQL, and then click Edit. The platform parameters appear.
+
+![Select Platform](Images/platfom.png)
+
+4. In the Automatic Password Management —> General parameters Section, make the following change.
+       
+       - Set the ImmediateInterval parameter to 1. (The number of minutes that will elapse between when the user initiates an account management process and when the process is performed.)
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:35]: is the Immediate Interval meant to be one word?

Comment (jkitchens32) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:35]: Maybe if so we can add the `` around it so it formats differently.

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+
+![Change Password](Images/showpass2.png)
+
+2. On the Activities screen, Click the CPM Change Password action to display details about the password change process.
+
+## Login to EDB Postgres Advanced Server with PSQL Client using the new Password
+
+The following steps demonstrate that user can login to the EDB Postgres Advanced Server with the changed password.
+
+1. Open PSQL client and login to EDB Postgres Advanced Server with the changed password. Login will be successful.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:103]: For this I would say something like "Login to EDB Postgres Advanced Server psql environment with the changed password.  This example uses a PSQL client on Windows." or something to that effect.

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+![Change Password](Images/login1.png)
+
+2. Try login with another password, error will be displayed.
```
Comment (jkitchens32) [inline advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:107]: Try to login with another password, the login will fail and an error message will be displayed.

**Review by jkitchens32 — CHANGES_REQUESTED** [review]

File: advocacy_docs/partner_docs/Cyberark/03-SolutionSummary.mdx (hunk 1 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/03-SolutionSummary.mdx:10]: Done

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx (hunk 2 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:8]: Removed "components".

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx (hunk 3 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:12]: done

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx (hunk 5 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:16]: done

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx (hunk 4 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:14]: Testing is done for EPAS only so mentioned EPAS here, same is the case with Connect CDC guids as well

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx (hunk 6 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:25]: done

File: advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx (hunk 7 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/04-Configuratingcyberark.mdx:42]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 8 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:6]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 9 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:8]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 10 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:10]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 11 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:15]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 13 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:19]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 14 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:35]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 12 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:18]: done

File: advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx (hunk 16 above)
Comment (kashifzeeshan) [reply advocacy_docs/partner_docs/Cyberark/05-Usingcyberark.mdx:107]: done
